### PR TITLE
Cp node labels

### PR DESCRIPTION
**Pull request by danbudris**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/vsphere/vsphere.go [hunk 1]
```diff
@@ -663,6 +663,7 @@ func buildTemplateMapCP(clusterSpec *cluster.Spec, datacenterSpec v1alpha1.VSphe
 	format := "cloud-config"
 	etcdExtraArgs := clusterapi.SecureEtcdTlsCipherSuitesExtraArgs()
 	sharedExtraArgs := clusterapi.SecureTlsCipherSuitesExtraArgs()
+	kubeletExtraArgs := clusterapi.ControlPlaneNodeLabelsExtraArgs(clusterSpec.Spec.ControlPlaneConfiguration).Append(sharedExtraArgs)
```
Comment (g-gaston) [inline pkg/providers/vsphere/vsphere.go:666]: Test for this?

**Review by taneyland — APPROVED** [review]
Looks great!
/lgtm

File: pkg/clusterapi/extraargs_test.go [hunk 2]
```diff
@@ -336,8 +336,42 @@ func TestNodeLabelsExtraArgs(t *testing.T) {
...
+			want: clusterapi.ExtraArgs{
+				"node-labels": "label1=foo,label2=bar",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.testName, func(t *testing.T) {
+			if got := clusterapi.ControlPlaneNodeLabelsExtraArgs(tt.cpc); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("WorkerNodeLabelsExtraArgs() = %v, want %v", got, tt.want)
```
Comment (abhay-krishna) [inline pkg/clusterapi/extraargs_test.go:374]:
```suggestion
				t.Errorf("ControlPlaneNodeLabelsExtraArgs() = %v, want %v", got, tt.want)
```

**Review by abhay-krishna — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
/override eks-anywhere-e2e-presubmit 
/override eks-anywhere-presubmit

This PR passed these already, they got retriggered after the nit fix.

**Pull request merged**